hw/ip/aon_timer/doc/theory_of_operation.md

```diff
@@ -1,25 +1,35 @@
 # Theory of Operation
 
-## Block Diagram
+AON Timer interacts with the CPU core and the power manager and reset manager to generate interrupts as well as wakeup and  reset events.
+Each timer increases its count as long as its enable signal (`wdog_ctrl.enable` for WDOG and `wkup_ctrl.enable` for WKUP) is set.
+Both timers halt the counting if the signal `lc_escalate_en_i` is set, which the alert handler is the system is in a "killed" state.
+In addition, the WDOG timer can be configured to stop counting if the `sleep_mode_i` is set.
+SW needs to set the field `wdog_ctrl.pause_in_sleep` to 1 in order to enable stopping.
+This feature is used to stop the watchdog timer running when in debugging mode.
 
-![AON Timer Block Diagram](../doc/aon_timer_block_diagram.svg)
+The diagram below depicts high level functionality and connectivity between AON Timer, the PLIC, Power Manager and the Reset Manager.
+The output signals `intr_wkup_timer_expired_o`, `intr_wdog_timer_bark_o` and `nmi_wdog_timer_bark_o` are driven on the SYS domain.
+The output signals `wkup_req_o` and `aon_timer_rst_req_o` are driven on the AON domain.
+Note the signal `nmi_wdog_timer_bark_o` is a copy of `intr_wdog_timer_bark_o`.
+
+### Block diagram
 
-See the block diagram for high level function and connectivity.
-The timer interacts with the CPU core and the power manager and reset manager to drive wakeup / reset events and interrupts.
-There is also an extra input to tell the counter whether to run ("counter-run").
-This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a "killed" state.
+![AON Timer Block Diagram](../doc/aon_timer_block_diagram.svg)
 
 ## Design Details
 
 The always-on timer will run on a ~200KHz clock.
-The wakeup timer is 64b wide and the watchdog timer 32b wide.
+The wakeup timer is 64-bit wide and the watchdog timer is 32-bit wide.
```

Comment:
Super nit: I think you want e.g. "64 bits wide" here. The hyphenated form gives an adjective, so the grammar expects "The 32-bit bus has 32 bits."